EnumObjectTypes.js

```diff
@@ -1,82 +1,106 @@
 /**
  *
- * Enumerate object types from nt!ObpRootDirectoryObject
+ * Enumerate all objects in nt!ObpRootDirectoryObject
  *
  */
 "use strict";
 
 
 const log = x => host.diagnostics.debugLog(x + "\n");
-
+const getHeader = x => x.address.subtract(host.getModuleType("nt", "_OBJECT_HEADER").fields.Body.offset);
+const getName = x  => host.createTypedObject(getHeader(x), "nt", "_OBJECT_HEADER").ObjectName
+const getTypeName = x  => host.createTypedObject(getHeader(x), "nt", "_OBJECT_HEADER").ObjectType
 
 /**
  *
  */
-function *ListObjects()
+function *DumpDirectory(objectDirectory, parentName)
 {
-
-    var ptrsz = host.namespace.Debugger.State.PseudoRegisters.General.ptrsize;
-    var hdrsz = host.evaluateExpression("sizeof(_OBJECT_HEADER)");
-
     //
-    // We must substract because _OBJECT_HEADER and the object body itself overlap by sizeof(void*)
-    // http://codemachine.com/article_objectheader.html
+    // Create the full directory name
     //
-    var hdroff = hdrsz - ptrsz;
+    var rootName = parentName + "\\" + getName(objectDirectory).slice(1, -1);
+    if (rootName === "\\\\") rootName = "";
 
     //
-    // x nt!ObpRootDirectoryObject
-    //
-    var pObpRootDirectoryObject = host.getModuleSymbolAddress("nt", "ObpRootDirectoryObject");
```

Comment:
I noticed in your scripts that you often get the address of a symbol, and then try to cast it to a pointer object of that symbol.

This is actually error-prone (for example, in this case, you were not actually getting ObpRootDirectoryObject but rather dereferencing it once.

The much easier way is to simply use getModuleSymbol as I did lower in this script, which lets you basically combine the two together (and correctly).